### ISSUE TITLE
Blazor circuit errors should be record as 'Error' rather than 'Debug' by the logging 

### ISSUE THREAD

**Issue by tomRedox** [aspnet/AspNetCore#17508]
I'm adding Blazor components into an existing MVC application.

My solution contains:
 - A server side Blazor project
 - An MVC project
 - lots of other library projects

All running on .NET Core 3.1 Preview 2 or .NET Standard

I place the components from the Blazor project on my MVC pages like this:

```c#
@(await Html.RenderComponentAsync<MyComponent>(RenderMode.Server))
```
 
If that component fails during the [Component Instantiation](https://docs.microsoft.com/en-us/aspnet/core/blazor/handle-errors?view=aspnetcore-3.0#component-instantiation) phase (for example in my case because my component was trying to access the HTTP context) the application appears to fail silently, that is to say:
 - No entry is shown in the browser console
 - No entry is shown on the web-server's console in VS

The MVC page will still finish loading, but the Blazor component is then missing for seemingly no reason.

It turns out that this failure *is* actually recorded, it's logged by the built in logging, but it's logged at the "Debug" level rather than as an "Error".

This seems too easy to miss to me, and also it doesn't match up to the error severity you get if you try to run a Blazor project with a missing dependency error, as in that case the app won't start and you'll immediately get an error message.

Is it possible to change the logging level for circuit issues?

Once a decision is made on this I'd plan to stick in a PR to make this slightly clearer in the documentation in one or both of these sections:
https://docs.microsoft.com/en-us/aspnet/core/blazor/handle-errors?view=aspnetcore-3.1#component-instantiation
https://docs.microsoft.com/en-us/aspnet/core/blazor/handle-errors?view=aspnetcore-3.1#prerendering
I'm probably an edge-case at the moment, I don't think there are that many people putting SSB components into MVC/Razor pages, but I expect the numbers will increase.

**Comment by javiercn** [comment]
@tomRedox thanks for contacting us.

I believe you should see a gold bar at the bottom of the page. 
@ryanbrandenburg do we not show this when a component fails to instantiate during development?

@danroth27 I think if we don't show a visible indicator, this is something we could do during development.

**Comment by tomRedox** [comment]
@javiercn should the gold bar work for a Blazor server side component that's running on an MVC page?  I thought the gold bar only applied for when the whole app was Blazor?

**Comment by javiercn** [comment]
@tomRedox those are not two different scenarios. A "blazor app" is just an MVC/RazorPages app rendering a single component that happens to represent an entire application.

**Comment by tomRedox** [comment]
@javiercn thanks, that's really useful.  

Knowing the gold bar should be showing got me a bit further along and made me realise that I hadn't added the markup for the gold bar to my app(!).  

Once I added the markup, the gold bar then started working for certain types of errors, but I'm still not getting gold bar errors in these scenarios when loading a MVC page with a SSB component:
 - When a dependency registration is missing for a service injected into that component
 - When an error occurs in the constructor for of a service injected into a component

I think that's probably a different issue to what I've written above, so I've made a new case for that on #17522.  I think solving #17522 will probably also resolve this case, so I'll close this one for now in favour of that one.
